src/main/groovy/io/cify/framework/core/Device.groovy

```diff
@@ -134,10 +136,18 @@ class Device implements IDevice {
         LOG.debug(MARKER, "Open app $app, $appActivity, $appPackage")
         try {
             File appFile = new File(app)
-            if (appFile.isFile() && (!app.startsWith("http://") && !app.startsWith("https://"))) {
+            if (appFile.isFile() && (!app.startsWith("http://") && !app.startsWith("https://")
+                                                                && !app.startsWith("s3://"))) {
                 String fileName = appFile.getName()
                 String path = appFile.toURI().getRawPath().replace(fileName, "").replace(":", "").toLowerCase()
                 app = path + fileName
+
+                /* app path workaround works only if test executor OS and farm OS are the same type */
+                String osName = System.getProperty("os.name").toLowerCase()
+                if (osName.indexOf("win") >= 0 ) {
+                    app = app.substring(1, 2) + ":" + app.substring(2, app.length())
+                    app.replace("/", "\\")
+                }
             }
 
             app ? setCapability("app", app) : null
```

Comment:
Could you please add return statements as in runner?

Reply:
This particular method is void, but I've added return statements, where possible